send_telegram.py

```diff
@@ -0,0 +1,46 @@
+import requests
+import os
+from logger import logger
+from load_config import *
+
+config = load_config('config.yml')
+
+telegram_enabled = "ENABLED" in config['TELEGRAM'] and config['TELEGRAM']['ENABLED']
+
+if not telegram_enabled:
+   logger.info('Telegram is disabled')
+else:
+   if "BOT_TOKEN" in config['TELEGRAM'] and "BOT_CHAT_ID" in config['TELEGRAM']:
+      # read config
+      bot_token = config['TELEGRAM']['BOT_TOKEN']
+      bot_chatID = str(config['TELEGRAM']['BOT_CHAT_ID'])
+      logger.info('Telegram initialized')
+   else:
+      # bail when config is not supplied
+      logger.info('Telegram not configured -> disabled')
+      telegram_enabled = False
+
+
+def send_telegram(message, key = 'DEBUG'):
+   """
+   send_telegram sends a notification message to telegram
+
+   :param message: the message to send
+   :param key: message key, allows enabling/disabling of specific messages, defaults to DEBUG
+   """ 
+
+   # telegram is disabled, don't do anything
+   if not telegram_enabled:
+      return
+
+   if key in config['TELEGRAM']['NOTIFICATIONS']:      
+      if config['TELEGRAM']['NOTIFICATIONS'][key]:
+         logger.debug(f'TELEGRAM sending {key} {message}')
```

Comment:
You log all those messages right before each `send_telegram` call which makes them redundant. Especially since some messages can be quite big I'd recommend only logging them once and focus on sending the telegram in this function.